models/project.py

```diff
@@ -0,0 +1,64 @@
+from google.appengine.ext import ndb
+
+
+class Project(ndb.Model):
+
+    title = ndb.StringProperty()
+    member_keys = ndb.KeyProperty(repeated=True, kind='User')
+    task_keys = ndb.KeyProperty(repeated=True, kind='Task')
+    course = ndb.KeyProperty()
+
+    @property
+    def members(self):
+        return map(lambda k: k.get(), self.member_keys)
+
+    def tasks(self):
+        return map(lambda k: k.get(), self.task_keys)
+
+    @property
+    def url(self):
+        return '/project?key={}'.format(self.key.urlsafe())
+
+    @classmethod
+    def new(cls, title):
```

Comment:
Yeah. I went back and forth on both of those features:
1. using new--potentially thinking of looking up projects similarly to classes but scaled down the method rather than totally removing it, to add organizing it differently
2. constructing with course key or not. clarified by the fact that the user needs to have their individual projects organized by class/queried that way, it still make sense. Sorry for the syntactical error there with an intermediate state.